hooks/hiero_update_shot.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2018 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+#  UI Hook
+# ===========================
+
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class HieroUpdateShot(HookBaseClass):
+    """
+    This class defines methods that handle updating the Shot entity
+    in Shotgun, as well as whether and how the filesystem structure
+    is created for a Shot during export.
+    """
+    def create_filesystem_structure(self, entity_type, entity_id, preset_properties):
+        """
+        Handles creating the filesystem structure for the shot that
+        was exported. The preset properties dictionary is provided to
+        allow for the lookup of any custom properties that might have
+        been defined in other hooks, like can be achieved when using
```

Comment:
a practical example to illustrate here would be great.

Reply:
Done.